src/hotspot/share/code/codeBlob.cpp

```diff
@@ -274,23 +286,15 @@ void* BufferBlob::operator new(size_t s, unsigned size) throw() {
 }
 
 void BufferBlob::free(BufferBlob *blob) {
-  assert(blob != NULL, "caller must check for NULL");
-  ThreadInVMfromUnknown __tiv;  // get to VM state in case we block on CodeCache_lock
-  blob->flush();
-  {
-    MutexLocker mu(CodeCache_lock, Mutex::_no_safepoint_check_flag);
-    CodeCache::free((RuntimeBlob*)blob);
-  }
-  // Track memory usage statistic after releasing CodeCache_lock
-  MemoryService::track_code_cache_memory_usage();
+  RuntimeBlob::free(blob);
 }
 
 
 //----------------------------------------------------------------------------------------------------
 // Implementation of AdapterBlob
 
 AdapterBlob::AdapterBlob(int size, CodeBuffer* cb) :
-  BufferBlob("I2C/C2I adapters", sizeof(AdapterBlob), size, cb) {
+  BufferBlob("I2C/C2I adapters", size, cb) {
```

Comment:
the header size here used to be sizeof(AdapterBlob), but now is sizeof(BufferBlob) - is that an issue?

Reply:
This is not an issue since they are the same. Arguably the header size should be passed as `sizeof(AdapterBlob)` for clarity, but I've reverted this to how it is done in the mainline jdk to avoid unnecessary churn for the next JEP: https://github.com/openjdk/jdk/blob/master/src/hotspot/share/code/codeBlob.cpp#L292-L295